backoff.go

```diff
@@ -57,8 +58,8 @@ func NewExponentialBackoff(initialTimeout, maxTimeout time.Duration, exponentFac
 
 // Next returns next time for retrying operation with exponential strategy
 func (eb *exponentialBackoff) Next(retry int) time.Duration {
-	if retry <= 0 {
-		return 0 * time.Millisecond
+	if retry < 0 {
```

Comment:
Is this to check for negative values for retry?

Reply:
Yea this shouldn't make any difference, since the loop in Client.Do always starts with 0, so this is just being defensive against possible future uses of the function